unlock-app/src/components/content/EventContent.tsx

```diff
@@ -7,20 +7,39 @@ import { AppLayout } from '../interface/layouts/AppLayout'
 import LoadingIcon from '../interface/Loading'
 import EventDetails from './event/EventDetails'
 import { EventLandingPage } from './event/EventLandingPage'
+import { useGetLockSettingsBySlug } from '~/hooks/useLockSettings'
 
 export const EventContent = () => {
   const router = useRouter()
-  if (!router.query) {
-    return <LoadingIcon></LoadingIcon>
-  }
 
-  const { lockAddress, network } = router.query
-  const showDetails = lockAddress && network
+  const { s: slug } = router.query
+  const {
+    isFetching,
+    isLoading,
+    data: lockSettings,
+  } = useGetLockSettingsBySlug(slug as string)
+
+  const loading = isFetching && isLoading
 
   const handleCreateEvent = () => {
     router.push('/event/new')
   }
 
+  if (!router.query || loading) {
+    return <LoadingIcon />
+  }
+
+  const lockAddress = lockSettings
+    ? lockSettings.lockAddress
+    : (router.query.lockAddress as string)
+
+  const network = lockSettings
+    ? lockSettings.network
+    : parseInt(router.query?.network?.toString() as string, 10)
+
+  const showDetails =
+    (router.query?.lockAddress && router.query?.network) || (slug && !loading)
```

Comment:
Let's just move to a helper hooks `useRouterQueryForLockAddressAndNetworks` whichg just yields `lockAddress` and `network`, whether it got it from a slug or the query params so that it is cleaner (you can then move the logic of `useGetLockSettingsBySlug` inside of it)